libs/util/git/src/git-client.service.ts

```diff
@@ -149,31 +151,41 @@ export class GitClientService {
     );
 
     try {
-      const { defaultBranch } = await this.provider.getRepository({
-        owner,
-        repositoryName,
-        groupName: repositoryGroupName,
-      });
+      let baseBranch: string;
+
+      //if not base branch name is provided, use the default branch of the repository
+      if (isEmpty(baseBranchName)) {
+        const repo = await this.provider.getRepository({
+          owner,
+          repositoryName,
+          groupName: repositoryGroupName,
+        });
+        baseBranch = repo.defaultBranch;
+      } else {
+        baseBranch = baseBranchName;
+      }
 
       await gitCli.clone();
 
-      const firstCommitOnDefaultBranch = await gitCli.getFirstCommitSha(
-        defaultBranch
+      const firstCommitOnBaseBranch = await gitCli.getFirstCommitSha(
+        baseBranch
       );
 
-      if (!firstCommitOnDefaultBranch) {
+      //This is the first commit by Amplication on the base branch
```

Comment:
No commits at all on base branch, regardless of Amplication

Reply:
no commit at all on the base branch - meaning this is the first commit by implication on the base branch :-)